spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -1105,7 +1105,8 @@ object CometSparkSessionExtensions extends Logging {
     try {
       // This will load the Comet native lib on demand, and if success, should set
       // `NativeBase.loaded` to true
-      NativeBase.isLoaded
+      NativeBase.load()
```

Comment:
I'm surprised we have to do this. I always understood that the static initializer would be called before we can invoke a static method.

Reply:
The goal of this PR was to remove the static initializer and explicitly load the native library so that we can catch exceptions and report them.